### PR TITLE
Enable serialization of JS code events and extension descriptions as arrays of strings

### PR DESCRIPTION
**Pull request by D8H**
… to make collaboration/reviews easier

Only show in developer changelog

It follows:
- https://github.com/4ian/GDevelop/pull/4468

Some changes on the extensions scripts:
- https://github.com/GDevelopApp/GDevelop-extensions/pull/679
- https://github.com/GDevelopApp/GDevelop-extensions/pull/680
- https://github.com/GDevelopApp/GDevelop-extensions/pull/681

Tested on:
- https://github.com/GDevelopApp/GDevelop-extensions/pull/669

### CODE REVIEW

File: Core/GDCore/Serialization/SerializerElement.cpp [hunk 1]
```diff
@@ -290,8 +290,8 @@ void SerializerElement::Init(const gd::SerializerElement& other) {
 }
 
 // TODO Remove shouldAlwayUseSetString in a release that follows the 150.
```
Comment (AlexandreSi) [inline Core/GDCore/Serialization/SerializerElement.cpp:292]: TODO to remove?

**Comment by D8H** [comment]
It can't be merged because of an issue with the extension description attribute:
- https://github.com/GDevelopApp/GDevelop-extensions/pull/682

**Comment by 4ian** [comment]
> A change was needed in the extension registry for the description in extension headers.

Isn't this necessary for JS events too?

**Comment by D8H** [comment]
> > A change was needed in the extension registry for the description in extension headers.
> 
> Isn't this necessary for JS events too?

The issue was for headers that are used by newIDE directly for the install dialog. The extension itself is parsed by Core so it uses your code.

**Review by 4ian — APPROVED** [review]

**Pull request merged**